lib/ui/painting.dart

```diff
@@ -4063,6 +4063,53 @@ abstract class ImageFilter {
     return _ComposeImageFilter(innerFilter: inner, outerFilter: outer);
   }
 
+  /// Creates an image filter from a [FragmentShader].
+  ///
+  /// The fragment shader provided here has additional requirements to be used
+  /// by the engine for filtering. The first uniform value must be a vec2, this
+  /// will be set by the engine to the size of the bound texture. There must
+  /// also be at least one sampler2D uniform, the first of which will be set by
+  /// the engine to contain the filter input.
+  ///
+  /// For example, the following is a valid fragment shader that can be used
+  /// with this API. Note that the uniform names are not required to have any
+  /// particular value.
+  ///
+  /// ```glsl
+  /// #include <flutter/runtime_effect.glsl>
+  ///
+  /// uniform vec2 u_size;
+  /// uniform float u_time;
+  ///
+  /// uniform sampler2D u_texture_input;
+  ///
+  /// out vec4 frag_color;
+  ///
+  /// void main() {
+  ///   frag_color = texture(u_texture_input, FlutterFragCoord().xy / u_size) * u_time;
+  ///
+  /// }
+  ///
+  /// ```
+  ///
+  /// This API is only supported when using the Impeller rendering engine. On
+  /// other backends a [UnsupportedError] will be thrown. To check at runtime
+  /// whether this API is suppored use [isShaderFilterSupported].
+  factory ImageFilter.shader(FragmentShader shader) {
+    if (!_impellerEnabled) {
+      throw UnsupportedError('ImageFilter.shader only supported with Impeller rendering engine.');
+    }
+    if (shader._floats.length < 2 || !shader._validateImageFilter()) {
```

Comment:
Can this error message be more apropos to what they did wrong.  For example if they are missing just the sampler, can we just have the error talk about the sampler?  This will save developers time as they respond to the problem.